### PR TITLE
Optimize images, Update azure-pipelines.yml, AppInstallerCLIE2ETests.csproj, and Microsoft.Management.Configuration.UnitTests.csproj

### PR DESCRIPTION
**Pull request by skanda890**
- [X] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).
- [ ] This pull request is related to an issue.

-----
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/4727)

### CODE REVIEW

File: package-lock.json
Comment (Trenly) [inline package-lock.json:1]: What is the reason for adding this file?

File: azure-pipelines.yml
Comment (Trenly) [inline azure-pipelines.yml:1]: I don't think that this job would be doing anything, and probably isn't correct?

File: package-lock.json
Comment (skanda890) [reply package-lock.json:1]: The `package-lock.json` file in npm serves several important purposes:

- **Dependency Locking**: It locks down the specific versions of every installed package, ensuring that the same versions are used whenever the project is installed, preventing unintended updates.
- **Version Consistency**: Ensures that every developer working on the project, as well as CI/CD systems, use the same versions of dependencies. This helps avoid the "it works on my machine" problem.
- **Improved Installation Speed**: Optimizes dependency installation by storing a flat `node_modules` structure, reducing the need for deep dependency resolution during installation.
- **Detailed Dependency Tree**: Provides a detailed, deterministic record of the dependency tree, making it easier to track and manage dependencies.
- **Facilitates Time-Travel**: Allows users to "time-travel" to previous states of `node_modules` without having to commit the directory itself.

Comment (mdanish-kh) [reply package-lock.json:1]: @skanda890 winget-cli is a C++ project, and doesn't use `npm`, so this file would serve no purpose here

File: azure-pipelines.yml
Comment (skanda890) [reply azure-pipelines.yml:1]: I think I fixed it.

**Comment by skanda890** [comment]
@Trenly, any more suggestions?

**Comment by Trenly** [comment]
It looks like you've done something weird again where you've merged several other branches into your master branch, which is causing 78 files to be modified, many which don't need to be modified such as `expect.txt`

**Comment by skanda890** [comment]
> It looks like you've done something weird again where you've merged several other branches

Microsoft/Winget-CLI or skanda890/Winget-CLI branches.

**Comment by skanda890** [comment]
> It looks like you've done something weird again where you've merged several other branches into your master branch, which is causing 78 files to be modified, many of which don't need to be modified such as `expect.txt`

'expect.txt' was told by Github.

**Comment by skanda890** [comment]
> It looks like you've done something weird again where you've merged several other branches into your master branch, which is causing 78 files to be modified, many which don't need to be modified such as `expect.txt`

Now what can I do?

**Pull request closed without merging**